src/System.Resources.Reader/src/System/Resources/ResourceReader.cs

```diff
@@ -50,31 +45,31 @@ public void Dispose()
         [System.Security.SecuritySafeCritical]  // auto-generated
         private unsafe void Dispose(bool disposing)
         {
-            if (_store != null)
+            if (_stream != null)
             {
 
                 if (disposing)
                 {
                     // Close the stream in a thread-safe way.  This fix means 
                     // that we may call Close n times, but that's safe.
-                    BinaryReader copyOfStore = _store;
-                    _store = null;
-                    if (copyOfStore != null)
-                        copyOfStore.Dispose();
+                    Stream stream = _stream;
```

Comment:
BTW, I as I was looking at the code in general, I noticed that it locks the stream in some members but not the others. I think in general this type is not thread safe and I thought about removing all the locks. What do you think? As part of this I would clean the comments as well.